client/src/main/java/org/moparforia/client/Launcher.java

```diff
@@ -49,6 +55,42 @@ public static boolean isUsingCustomServer() {
         return true;//instance.serverBox.isSelected();
     }
 
+    public void loadPreferences() {
```

Comment:
I would move all this Preferences logic into a separate class and add getters/setters for individual properties, this way we can handle all the logic for one property being unknown there. It will be also easier to test, since we could be mocking  individual methods in that new class instead of the entire properties.

Reply:
Getters could return Optional and the individual getters could work like this:
* Is the property in loaded properties?
   * Yes, return the property
   * Not, return empty Optional